common/flatpak-dir.c

```diff
@@ -213,6 +215,126 @@ flatpak_get_system_default_base_dir_location (void)
   return g_file_new_for_path ((char *)path);
 }
 
+static void
+append_locations_from_config_file (GPtrArray *locations, const char *file_path)
+{
+  g_autoptr(GKeyFile) keyfile = NULL;
+  g_autoptr(GError) error = NULL;
+  g_auto(GStrv) groups = NULL;
+  gsize n_groups;
+  int i;
+
+  keyfile = g_key_file_new ();
+
+  g_key_file_load_from_file (keyfile, file_path, G_KEY_FILE_NONE, &error);
+  if (error != NULL)
+    {
+      g_warning ("Could not get list of system installations: %s\n", error->message);
+      return;
+    }
+
+  /* One configuration file might define more than one installation */
+  groups = g_key_file_get_groups (keyfile, &n_groups);
+  for (i = 0; i < n_groups; i++)
+    {
+      g_autofree char *id = NULL;
+      g_autofree char *path = NULL;
+      size_t len;
+
+      if (!g_str_has_prefix (groups[i], "Installation \""))
+        continue;
+
+      id = g_strdup (&groups[i][14]);
+      if (!g_str_has_suffix (id, "\""))
+        continue;
+
```

Comment:
Ok. But I will add this check in a later commit if that's ok, as it will be easier since at this point all I'm building is a list of paths